### PR TITLE
Speed up launcher startup

### PR DESCRIPTION
**Pull request by RebeccaMahany**
* `Once()` can take up to 5 seconds to run -- don't block on running it
* We can't use the Secure Enclave right now -- stop waiting for up to a second to access it

### CODE REVIEW

File: ee/agent/keys.go [hunk 1]
```diff
@@ -39,18 +40,21 @@ func SetupKeys(logger log.Logger, store types.GetterSetterDeleter) error {
 		return fmt.Errorf("setting up local db keys: %w", err)
 	}
 
-	err = backoff.WaitFor(func() error {
-		hwKeys, err := setupHardwareKeys(logger, store)
+	// Secure Enclave is not currently supported, so don't spend startup time waiting for it to work -- see keys_darwin.go for more details.
+	if runtime.GOOS != "darwin" {
```
Comment (zackattack01) [inline ee/agent/keys.go:44]: might be simpler as `if runtime.GOOS == "darwin" { return nil }`

File: cmd/launcher/launcher.go [hunk 2]
```diff
@@ -276,8 +276,9 @@ func runLauncher(ctx context.Context, cancel func(), multiSlogger, systemMultiSl
 	}
 
 	// Add the log checkpoints to the rungroup, and run it once early, to try to get data into the logs.
+	// The checkpointer can take up to 5 seconds to run, so do this in the background.
 	checkpointer := checkups.NewCheckupLogger(logger, k)
-	checkpointer.Once(ctx)
+	go checkpointer.Once(ctx)
```
Comment (zackattack01) [inline cmd/launcher/launcher.go:281]: nice catch! 🔥

**Review by zackattack01 — DISMISSED** [review]

**Review by James-Pickett — DISMISSED** [review]
nice

File: ee/agent/keys.go (hunk 1 above)
Comment (RebeccaMahany) [reply ee/agent/keys.go]: Ah yup that reads much better, thanks -- updated

File: cmd/launcher/launcher.go (hunk 2 above)
Comment (directionless) [inline cmd/launcher/launcher.go:281]: The prior way was intentional -- it was intending to get some of the identifying information into the logs _before_ any early crashes might occur. It's fine to revisit, maybe we don't need it with all the things we've subsequently added.

Though I think this begs the question -- why does it take 5s to log, and should we add go routines downstack?

File: ee/agent/keys.go [hunk 3]
```diff
@@ -39,6 +40,11 @@ func SetupKeys(logger log.Logger, store types.GetterSetterDeleter) error {
 		return fmt.Errorf("setting up local db keys: %w", err)
 	}
 
+	// Secure Enclave is not currently supported, so don't spend startup time waiting for it to work -- see keys_darwin.go for more details.
+	if runtime.GOOS == "darwin" {
+		return nil
+	}
```
Comment (directionless) [inline ee/agent/keys.go:46]: @James-Pickett Is this okay with the other work you have in flight?

File: cmd/launcher/launcher.go (hunk 2 above)
Comment (RebeccaMahany) [reply cmd/launcher/launcher.go:281]: As suspected -- it's the connectivity checkup: https://github.com/kolide/launcher/blob/main/ee/debug/checkups/connectivity.go

Comment (directionless) [reply cmd/launcher/launcher.go:281]: I wonder if we should just async that one check. Not sure

Comment (RebeccaMahany) [reply cmd/launcher/launcher.go:281]: That seemed like maybe overly complicated to coordinate an exclusion for that one checkup for the checkpoint only, but not doctor/flare

Comment (directionless) [reply cmd/launcher/launcher.go:281]: Thinking about it... Maybe this simple thing is good enough for now.

We probably _do_ want to keep this in the log checkpoint, it provides an interesting point-in-time view. 

I suspect that the theoretical correct state would be to make the timer log checkpoints use a different set of debuggers than the initial startup ones. And I'm not sure it's worth doing that this moment

Comment (RebeccaMahany) [reply cmd/launcher/launcher.go:281]: Agreed!

**Pull request merged**